### PR TITLE
Add is.TupleLike

### PR DESCRIPTION
**Pull request by Myfeetarefreezing**
https://github.com/sindresorhus/is/issues/156 used this as reference to the interface

### CODE REVIEW

**Comment by sindresorhus** [comment]
You need to update the readme.

**Comment by Myfeetarefreezing** [comment]
Hope the `.md` is clear enough

**Comment by sindresorhus** [comment]
Thanks. Can you also add a few type tests to ensure the types are correctly narrowed?

Example: type ResolveTypesOfTypeGuardsTuple<TypeGuardsOfT> = {
  [K in keyof TypeGuardsOfT]: TypeGuardsOfT[K] extends TypeGuard<infer U> ? U : never;
};

**Comment by sindresorhus** [comment]
Thanks. Can you also add a few type tests to ensure the types are correctly narrowed?

Example: https://github.com/sindresorhus/is/blob/278e0e9696b9862381a6e3cae34f756c67ef735b/test/test.ts#L1567-L1575

File: test/test.ts [hunk 1]
```diff
@@ -1480,6 +1480,47 @@ test('is.tupleLike', t => {
...
+		// eslint-disable-next-line @typescript-eslint/ban-types
+		const function_ = (value: number | string | boolean | undefined | null) => value;
+
+		if (is.tupleLike(tuple, [is.number, is.string, is.boolean, is.undefined, is.null_])) {
+			const numericValue = tuple[0];
+			const stringValue = tuple[1];
+			const booleanValue = tuple[2];
+			const undefinedValue = tuple[3];
+			const nullValue = tuple[4];
+			function_(numericValue);
```
Comment (sindresorhus) [inline test/test.ts:1517]: Make a function for each type, to make it more specific.

Comment (sindresorhus) [reply test/test.ts:1517]: Actually, just add https://github.com/mmkal/expect-type as a dev dependency and make proper type checks.

Comment (Myfeetarefreezing) [reply test/test.ts:1517]: on it

File: source/index.ts [hunk 2]
```diff
@@ -326,6 +326,19 @@ export type ArrayLike<T> = {
 const isValidLength = (value: unknown): value is number => is.safeInteger(value) && value >= 0;
 is.arrayLike = <T = unknown>(value: unknown): value is ArrayLike<T> => !is.nullOrUndefined(value) && !is.function_(value) && isValidLength((value as ArrayLike<T>).length);
 
+type TypeGuard<T> = (value: unknown) => value is T;
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+type ResolveTypesOfTypeGuardsTuple<TypeGuardsOfT, ResultOfT extends unknown[] = [] > = TypeGuardsOfT extends [TypeGuard<infer U>, ...infer TOthers] ? ResolveTypesOfTypeGuardsTuple<TOthers, [...ResultOfT, U]> : TypeGuardsOfT extends undefined[] ? ResultOfT : never;
+
+is.tupleLike = <T extends Array<TypeGuard<unknown>>>(value: unknown, guards: [...T]): value is ResolveTypesOfTypeGuardsTuple<T> => {
```
Comment (gioragutt) [inline source/index.ts:334]:
```suggestion
is.tupleLike = <T extends TypeGuard<unknown>[]>(value: unknown, guards: [...T]): value is ResolveTypesOfTypeGuardsTuple<T> => {
```

**Review by gioragutt — APPROVED** [review]
LGTM, GJ @Myfeetarefreezing, hope your feet get warmer 👍🏻

**Comment by Myfeetarefreezing** [comment]
> LGTM, GJ @Myfeetarefreezing, hope your feet get warmer 👍🏻

A never ending struggle..

**Comment by gioragutt** [comment]
Congrats @Myfeetarefreezing, thank you for your contribution 🙏🏻

**Pull request merged**